### PR TITLE
Fixed the OpenAL device not being properly closed

### PR DESCRIPTION
**Pull request by Mailaender**
Revival of https://github.com/OpenRA/OpenRA/pull/9395 as https://github.com/OpenRA/OpenRA/pull/9394 did not solve it completely.

### CODE REVIEW

File: OpenRA.Platforms.Default/OpenAlSoundEngine.cs [hunk 1]
```diff
@@ -102,6 +102,8 @@ public OpenAlSoundEngine()
 					throw new InvalidOperationException("Can't create OpenAL device");
 			}
 
+			Game.OnQuit += () => { ALC10.alcCloseDevice(device); };
```
Comment (pchote) [inline OpenRA.Platforms.Default/OpenAlSoundEngine.cs]: This is already done when disposing the sound engine.  If the sound engine isn't being disposed then that needs to be fixed – don't just force close it again!

Comment (Mailaender) [inline OpenRA.Platforms.Default/OpenAlSoundEngine.cs]: Shall I change this to Dispose OnQuit?

Comment (pchote) [inline OpenRA.Platforms.Default/OpenAlSoundEngine.cs]: This shouldn't be using OnQuit at all. That's a legacy hack from before we started treating object lifetimes properly.

**Comment by Mailaender** [comment]
Closing as this is not wanted and no constructive alternatives are provided. The bug https://github.com/OpenRA/OpenRA/issues/11193 persists.

**Comment by pchote** [comment]
I gave the alternative in my first comment: you need to identify the code path where the game quits without disposing the SoundDevice, and then add the code to dispose it.

**Comment by Mailaender** [comment]
No, you said:

> This is already done when disposing the sound engine.

**Comment by LipkeGu** [comment]
> If the sound engine isn't being disposed then that needs to be fixed – don't just force close it again!

**Comment by Mailaender** [comment]
Probably a misunderstanding, but either way I don't know why the game doesn't dispose properly when the window manager quits the game or ALT + F4 is hit.

**Comment by LipkeGu** [comment]
maybe it isnt a OpenRA related thing... mayb it is SDL2 related... we should forward this to the SDL devs :/

**Comment by LipkeGu** [comment]
i used google.... it is OpenAL related....
> 

The AL lib is the "audio library" used by Libgdx (one of the OpenAL variants).

I believe this message just means that the audio library is cleaning up some (in your case just one) audio streams/handles for you. If you see this at exit, its harmless as all resources will be cleaned up by the OS.

If you internally cleanup your audio before exiting, the message should go away.

For more details, look for alc_cleanup in here: http://repo.or.cz/w/openal-soft.git/blob/HEAD:/Alc/ALc.c


and also this info: AL.destroy(); must be called before the application shuts down. this call is normally included in a finally block, but if the application calls System.exit(0), finally blocks are no longer executed.

**Pull request closed without merging**